code/go/internal/pkgpath/files.go

```diff
@@ -66,20 +66,70 @@ func (f File) Values(path string) (interface{}, error) {
 		return nil, errors.Wrap(err, "reading file content failed")
 	}
 
-	var v interface{}
+	v := make(map[string]interface{})
 	if fileExt == "yaml" || fileExt == "yml" {
-		if err := yaml.Unmarshal(contents, &v); err != nil {
-			return nil, errors.Wrapf(err, "unmarshalling YAML file failed (path: %s)", f.fsys.Path(fileName))
+		config, err := yaml.NewConfig(contents)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing yaml file failed (path: %s)", f.fsys.Path(fileName))
 		}
+
+		config.Unpack(&v)
 	} else if fileExt == "json" {
-		if err := json.Unmarshal(contents, &v); err != nil {
-			return nil, errors.Wrapf(err, "unmarshalling JSON file failed (path: %s)", f.fsys.Path(fileName))
+		config, err := json.NewConfig(contents)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing json file failed (path: %s)", f.fsys.Path(fileName))
 		}
+		config.Unpack(&v)
 	}
 
 	return jsonpath.Get(path, v)
 }
 
+// ValuesArray Same as values but returns an array of values
+func (f File) ValuesArray(path string) ([]interface{}, error) {
```

Comment:
Would it be useful to check whether or not `fileName` is `changelog.yml` to just use one function? Depending on that maybe it could be unpack the config into one or another object type.